online-shop/src/Components/ShoppingCartComponent/ShoppingCartDumpView.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import '../../Styles/ComponentsStyles/ShoppingCartStyles/ShoppingCartStyles.scss';
+import { IProduct } from '../../Models/Models';
+import { ShoppingCartProps } from './ShoppingCartSmartView';
+import { ErrorComponent } from '../../HelperComponents/ErrorComponent';
+import { ActionStatusReport } from '../../HelperComponents/ActionStatusReport';
+
+const calculateNumberOfSameItem = (arrayToCalculateFrom: IProduct[], productId: number): number => {
+    let numberOfSameProduct = 0;
```

Comment:
You could do this more easily with "functional" programming: `arrayToCalculateFrom.filter(product => product.id === productId).length`.